Documentation/scalability/scalability.rst

```diff
@@ -0,0 +1,397 @@
+
+As expected, **increasing the number of policies does not have a significant
+impact in the memory usage of Cilium since the BPF policy maps have a constant
+size** once a pod is initialized.
+
+.. figure:: images/image_8_03.png
+.. figure:: images/image_8_04.png
+
+
+The first mark represents the point in time when we ran ``kubectl create`` to
+create the ``CiliumNetworkPolicies``. Since we created 250 policies sequentially
+we cannot correctly derive the time between the creation of the 250th CNP and
+the time the last endpoint had its policy enforced. In our examples we have
+used 250 different CNP objects, to correctly measure the convergence time we
+could have used a single CNP with multiple policy rules defined under the
+``specs`` field (not ``spec``) for a CNP.
+
+However we can see the time it took the last Cilium agent to increment its
+Policy Revision, which is incremented individually on each Cilium agent every
+time time a CiliumNetworkPolicy (CNP) is received, between second ``15:45:44``
+and ``15:45:46`` and see when was the last time an Endpoint was regenerated by
+checking the 99th percentile of the "Endpoint regeneration time", we can see
+that it took less than 5s. We can also verify **the maximum time was less than
+``600ms`` for an endpoint to have the policy enforced.**
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+9. Deploy 250 policies for CiliumClusterwideNetworkPolicies (CCNP)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The difference between these policies and the previous ones installed is that
+these ones select all pods in all namespaces. To recap, this means that we will
+now have **250 different network policies selecting 10000 pods and 250 different
+network policies selecting 50000 pods on a cluster with 1000 nodes.** Similarly
+to the previous step we will deploy 125 L4 policies and another 125 L7 policies.
+
+.. figure:: images/image_9_01.png
+.. figure:: images/image_9_02.png
+
+Similar to the creation of the previous 250 CNPs, there was also an increase in
+CPU usage during the creation of the CCNPs. The CPU usage was similar even
+though the policies were effectively selecting more pods.
+
+.. figure:: images/image_9_03.png
+
+As all pods running in a node are selected by **all the 250 CCNPs created**, we
+see an increase of the **Endpoint regeneration time** which **peaked a little
+above 3s.**
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+10. "Accidentally" destroy 10000 pods
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this step we have "accidentally" destroyed 10000 random pods. Kubernetes will
+then recreate 10000 new pods so it will help us understand what is the
+convergence time for all the network policies deployed.
+
+.. figure:: images/image_10_01.png
+.. figure:: images/image_10_02.png
+
+
+* The first mark represents the point in time when pods were "deleted" and the
+  second mark represents the point in time 10k new pods finished to be
```

Comment:
second mark represents the point in time Kubernetes finished recreating 10k pods.